### PR TITLE
chore(deps): update dependency jsonpath-plus to 10.0.0 due to vulnerability

### PR DESCRIPTION
**Pull request by knowacki23**
This is a simple dependency bump of the jsonpath-plus dependency.

This dependency is vulnerable according to Snyk: https://security.snyk.io/vuln/SNYK-JS-JSONPATHPLUS-7945884
As this is a non-major upgrade and I have not seen the vulnerability being mentioned anywhere in the repository I thought I'd open a PR for it.

### CODE REVIEW

**Comment by P0lip** [comment]
Could you please update the lockfile as well?
Also, it'd be good to change the commit title to `chore(deps)` as this change has no impact on the package code itself as mentiond [here](https://github.com/P0lip/nimma/issues/14#issuecomment-2412078857).
Nimma no longer relies on jsonpath-plus and we have it listed as a dev dependency only to use in test suites to compare Nimma results with those from jsonpath-plus.

**Comment by knowacki23** [comment]
Updated `package-lock.json` and updated commit and PR titles

File: package-lock.json [hunk 1]
```diff
@@ -785,20 +1015,26 @@
...
-      "license": "MIT"
+      "resolved": "https://registry.npmjs.org/json-stable-stringify-without-jsonify/-/json-stable-stringify-without-jsonify-1.0.1.tgz",
+      "integrity": "sha512-Bdboy+l7tA3OGW6FjyFHWkP5LuByj1Tk33Ljyq0axyzdk9//JSi2u3fP1QSmd1KNwq6VOKYGlAu87CisVir6Pw==",
+      "dev": true
     },
     "node_modules/jsonpath-plus": {
-      "version": "8.1.0",
-      "resolved": "https://registry.npmjs.org/jsonpath-plus/-/jsonpath-plus-8.1.0.tgz",
-      "integrity": "sha512-qVTiuKztFGw0dGhYi3WNqvddx3/SHtyDT0xJaeyz4uP0d1tkpG+0y5uYQ4OcIo1TLAz3PE/qDOW9F0uDt3+CTw==",
+      "version": "10.0.0",
```
Comment (johannesmarx) [inline package-lock.json:1023]: Hi @knowacki23, could you please update again? 
Your source reports that versions `< 10.0.7` are affected. Thanks a lot 👍

Comment (coreydaley) [reply package-lock.json:1023]: Wouldn't the semver `^10.0.0` pull in the newest `10.1.0` version? https://classic.yarnpkg.com/lang/en/docs/dependency-versions/#toc-caret-ranges

Comment (johannesmarx) [reply package-lock.json:1023]: To make the library build consistent and working with version `10.1.0`, `package-lock.json` should also be updated (again) to show explicitly that it's using a version `>= 10.0.7`.

**Comment by coreydaley** [comment]
@johannesmarx I opened #14 with the requested changes.
Based on another pull request from @knowacki23 (https://github.com/asyncapi/parser-js/pull/1056) I don't feel as though they are going to update this one.

**Comment by P0lip** [comment]
@knowacki23 thanks for the PR.
I decided to drop jsonpath-plus in https://github.com/P0lip/nimma/commit/17faf598e5d720971eb0a28ef2d28f4c75c90d6d.
It was only a dev dependency that wasn't really needed for anything other than convenience.

**Comment by coreydaley** [comment]
@P0lip Would you mind doing a new release now that it is removed?

**Comment by P0lip** [comment]
I don't think we need to.
Let me explain. Nimma wasn't vulnerable. jsonpath-plus was only a dev dependency that was used in one test suite.
In a typical setup, dev dependencies aren't installed when you install an npm package. In other words, when one adds nimma to their project, only astring (the only runtime dep Nimma requires) would be installed apart from nimma itself.
As such, the latest version of nimma should pass scanners that check for outdated/vulnerable dependencies.

**Pull request closed without merging**